### PR TITLE
SC-401 Poetry PM

### PR DESCRIPTION
**Pull request by epcAnker**
Special Things done:
-typedef struct GreenType_s GreenType_t; 
    ->
     typedef struct GreenType_s
    {
    } GreenType_t;, and added align = none
to get pycparser testing working
-venv -> .venv
-poetry publish instead of boots.
-modified versioning
-builduipm extra manual step

Validation:
image

### CODE REVIEW

**Review by gordon-epc — CHANGES_REQUESTED** [review]
Remove MANIFEST.in

**Review by gordon-epc — APPROVED** [review]

**Review by bholewin-epc — APPROVED** [review]

**Review by gordon-epc — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -30,20 +31,42 @@ jobs:
...
+      - name: Install Poetry and Venv
+        run: |
+          python3 -m ensurepip
+          python3 -m pip install --upgrade pip
+          python3 -m pip install poetry
+          poetry config virtualenvs.in-project true
+          poetry config virtualenvs.path .
+
+          #setup connection to aws server
+          poetry config repositories.epc-power https://epc-power-us-890711758583.d.codeartifact.us-east-2.amazonaws.com/pypi/epc-power/
```
Comment (gordon-epc) [inline .github/workflows/ci.yml:49]:
```suggestion
          poetry config repositories.epc-power https://epc-power-us-${{ secrets.AWS_ACCOUNT_ID }}.d.codeartifact.us-east-2.amazonaws.com/pypi/epc-power/
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -30,20 +31,42 @@ jobs:
...
+          export CODEARTIFACT_AUTH_TOKEN=`aws codeartifact get-authorization-token --domain epc-power-us --domain-owner 890711758583 --query authorizationToken --output text`
```
Comment (gordon-epc) [inline .github/workflows/ci.yml:50]:
```suggestion
          export CODEARTIFACT_AUTH_TOKEN=`aws codeartifact get-authorization-token --domain epc-power-us --domain-owner ${{ secrets.AWS_ACCOUNT_ID }} --query authorizationToken --output text`
```

**Review by gordon-epc — CHANGES_REQUESTED** [review]
Delete the file `DELETEtox.zip`

**Review by gordon-epc — APPROVED** [review]

**Pull request merged**